ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/DepositUtil.java

```diff
@@ -198,13 +196,10 @@ public static DepositWithIndex convertDepositEventToOperationDeposit(
     return new tech.pegasys.artemis.pow.event.Deposit(response);
   }
 
-  public static UnsignedLong getEpochBlockTimeByDepositBlockNumber(
-      BigInteger blockNumber, String provider) throws IOException {
+  public static UnsignedLong getEpochBlockTimeByDepositBlockHash(Bytes32 blockHash, String provider)
+      throws IOException {
     Web3j web3 = Web3j.build(new HttpService(provider));
     return UnsignedLong.valueOf(
-        web3.ethGetBlockByNumber(new DefaultBlockParameterNumber(blockNumber), true)
-            .send()
-            .getBlock()
-            .getTimestamp());
+        web3.ethGetBlockByHash(blockHash.toHexString(), true).send().getBlock().getTimestamp());
```

Comment:
Doesn't look like we need the transaction objects:
```suggestion
        web3.ethGetBlockByHash(blockHash.toHexString(), false).send().getBlock().getTimestamp());
```

Reply:
Good spot, we don't.